Sources/ApodiniHTTP/InterfaceExporter.swift

```diff
@@ -0,0 +1,408 @@
+    ///    - caseInsensitiveRouting: Indicates whether the HTTP route is interpreted case-sensitivly
+    public init(encoder: AnyEncoder = defaultEncoder, decoder: AnyDecoder = defaultDecoder, caseInsensitiveRouting: Bool = false) {
+        self.configuration = ExporterConfiguration(encoder: encoder, decoder: decoder, caseInsensitiveRouting: caseInsensitiveRouting)
+    }
+    
+    public func configure(_ app: Apodini.Application) {
+        /// Instanciate exporter
+        let exporter = Exporter(app, self.configuration)
+        
+        /// Insert exporter into `InterfaceExporterStorage`
+        app.registerExporter(exporter: exporter)
+    }
+}
+
+
+// MARK: Exporter
+
+struct Exporter: InterfaceExporter {
+    let app: Apodini.Application
+    
+    let configuration: HTTP.ExporterConfiguration
+    
+    let logger: Logger
+    
+    init(_ app: Apodini.Application, _ configuration: HTTP.ExporterConfiguration) {
+        self.app = app
+        self.configuration = configuration
+        var logger = app.logger
+        logger.logLevel = .trace
+        self.logger = logger
+        
+        // Set option to activate case insensitive routing, default is false (so case-sensitive)
+        self.app.vapor.app.routes.caseInsensitive = configuration.caseInsensitiveRouting
+    }
+    
+    static let parameterNamespace: [ParameterNamespace] = .individual
+    
+    func export<H>(_ endpoint: Endpoint<H>) where H: Handler {
+        let knowledge = endpoint[VaporEndpointKnowledge.self]
+        
+        switch knowledge.pattern {
+        case .requestResponse:
+            logger.info("Exporting Request-Response Pattern on \(knowledge.method): \(knowledge.path)")
+            app.vapor.app.on(knowledge.method,
+                             knowledge.path,
+                             use: buildRequestResponseClosure(for: endpoint, using: knowledge.defaultValues))
+        case .serviceSideStream:
+            logger.info("Exporting Service-Side-Streaming Pattern on \(knowledge.method): \(knowledge.path)")
+            app.vapor.app.on(knowledge.method,
+                             knowledge.path,
+                             use: buildServiceSideStreamingClosure(for: endpoint, using: knowledge.defaultValues))
+        case .clientSideStream:
+            logger.info("Exporting Client-Side-Streaming Pattern on \(knowledge.method): \(knowledge.path)")
+            app.vapor.app.on(knowledge.method,
+                             knowledge.path,
+                             use: buildClientSideStreamingClosure(for: endpoint, using: knowledge.defaultValues))
+        case .bidirectionalStream:
+            logger.info("Exporting Bidirectional-Streaming Pattern on \(knowledge.method): \(knowledge.path)")
+            app.vapor.app.on(knowledge.method,
+                             knowledge.path,
+                             use: buildBidirectionalStreamingClosure(for: endpoint, using: knowledge.defaultValues))
+        }
+    }
```

Comment:
I think this exporter is really showcases the benefits of this new way of building request handling strategies. With very little effort, and without the need for spaghetti code or a "hugely differing" code base you can support all the different communication patterns in a very composable style.
Chapeau, great work 🚀